### PR TITLE
[FILES] Allow user to delete or remove themselves from a shared folder

### LINKED ISSUE #1303: Files: Handle `Delete Shared Folder` for all users
**Is your feature request related to a problem? Please describe.**
An owner of a Shared Folder should be able to delete the shared folder.
A reader/writer that has been added to a shared folder should be allowed to remove themselves from the shared folder

**Describe the solution you'd like**
If the user is an `owner` of a shared folder, they would be displayed an option in the kebab menu to delete a bucket, followed by a confirmation modal warning them that all files will be deleted for all users. Once confirmed, the call to `DELETE {bucketId}` is made.

If a user is a `reader`/`writer`, there would be an option displayed in the kebab menu to remove themselves from the bucket, followed by a confirmation modal warning them that they will lose access to all the contents of the bucket. Once confirmed, the call to `DELETE {bucketId}` is made

### PR DESCRIPTION
**Pull request by FSM1**
Closes #1303

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx [hunk 1]
```diff
@@ -245,6 +267,25 @@ const SharedFolderOverview = () => {
...
+        }
+        rejectText = {t`Cancel`}
+        acceptText = {t`Confirm`}
+        acceptButtonProps={{ loading: isDeletingSharedFolder, disabled: isDeletingSharedFolder, testId: "confirm-deletion" }}
+        rejectButtonProps={{ disabled: isDeletingSharedFolder, testId: "cancel-deletion" }}
+        injectedClass={{ inner: classes.confirmDeletionDialog }}
+        onModalBodyClick={(e) => {
+          e.preventDefault()
+          e.stopPropagation()
+        }}
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx]: We won't need this here, only needed with grid view implementation
```suggestion
```

**Review by tanmoyAtb — APPROVED** [review]
Looks 💯 to me!

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 2]
```diff
@@ -205,7 +288,7 @@ const SharedFolderRow = ({ bucket, onFolderClick, menuItems, isEditing, setIsEdi
       </TableCell>
       {desktop &&
         <TableCell align="left">
-          {isOwner
+          {bucket.permission === "owner"
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: we should use `isOwner` here.
```suggestion
          {isOwner
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx [hunk 3]
```diff
@@ -245,6 +267,25 @@ const SharedFolderOverview = () => {
...
+        testId="file-deletion"
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/SharedFoldersOverview.tsx]: nit
```suggestion
        testId="shared-folder-deletion"
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 4]
```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
...
+    {
+      contents: (
+        <>
+          <UpdateSvg className={classes.menuIcon} />
+          <span data-cy="menu-edit">
+            <Trans>Edit</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => onEditSharedFolder()
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: nit as I see it.
```suggestion
      onClick: onEditSharedFolder
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 5]
```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
...
+    }, {
+      contents: (
+        <>
+          <DeleteSvg className={classes.menuIcon} />
+          <span data-cy="menu-delete">
+            <Trans>Delete</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => handleDeleteSharedFolder()
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: nit as I see it.
```suggestion
      onClick: handleDeleteSharedFolder
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 6]
```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
...
+    : [{
+      contents: (
+        <>
+          <DeleteSvg className={classes.menuIcon} />
+          <span data-cy="menu-delete">
+            <Trans>Leave</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => handleDeleteSharedFolder()
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: nit as I see it.
```suggestion
      onClick: handleDeleteSharedFolder
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 7]
```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
...
+
+  const onSingleClick = useCallback(
+    () => {
+      if (desktop) {
+        // on desktop 
+      } else {
+        // on mobile
+        openSharedFolder(bucket.id)
+      }
+    },
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: I know it's a copy/paste from me, but I feel we can get rid of this empty if.
```suggestion
      if (!desktop) {
        // on mobile
        openSharedFolder(bucket.id)
      }
    },
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx [hunk 8]
```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
...
+
+  const onDoubleClick = useCallback(
+    () => {
+      if (desktop) {
+        // on desktop
+        openSharedFolder(bucket.id)
+      } else {
+        // on mobile
+        return
+      }
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx]: Same here.
```suggestion
    () => {
      if (desktop) {
        // on desktop
        openSharedFolder(bucket.id)
      }
```

**Review by Tbaut — APPROVED** [review]
Working perfectly! Only have nits as I saw them, mostly not introduced by you be quick to fix :)

**Comment by asnaith** [comment]
Nice! Looking great on Render 👍

**Pull request merged**